standardtracer/propagation.go

```diff
@@ -127,7 +133,7 @@ func (p *splitBinaryPropagator) InjectSpan(
 	}
 
 	// Handle the trace and span ids, and sampled status.
-	contextBuf := new(bytes.Buffer)
+	contextBuf := bytes.NewBuffer(splitBinaryCarrier.TracerState[:0])
```

Comment:
this creates a buffer which uses the (truncated) slice from the binary carrier. That is, if the carrier has a 1000 character slice, we'll use the capacity of 1000 (but set the length to zero).